### PR TITLE
Unarchiver now handles an empty NAME input variable

### PR DESCRIPTION
**Pull request by nmcspadden**
This should address #433.

Using a very simple recipe with no NAME input variable:
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>Description</key>
	<string>Recipe description</string>
	<key>Identifier</key>
	<string>local.TestExtract</string>
	<key>Input</key>
	<dict/>
	<key>MiniumumVersion</key>
	<string>1.0</string>
	<key>Process</key>
	<array>
		<dict>
			<key>Arguments</key>
			<dict>
				<key>archive_path</key>
				<string>%PKG%</string>
			</dict>
			<key>Processor</key>
			<string>Unarchiver</string>
		</dict>
	</array>
</dict>
</plist>
```

Running it by passing in a zip file:
```
$ ./autopkg run TestExtract.extract -p ~/Downloads/test.zip
Processing TestExtract.extract...
WARNING: TestExtract.extract is missing trust info and FAIL_RECIPES_WITHOUT_TRUST_INFO is not set. Proceeding...
No destination path provided and NAME is empty, cannot pick a default unpack directory!
Failed.

The following recipes failed:
    TestExtract.extract
        Error in local.TestExtract: Processor: Unarchiver: Error: No destination path provided and NAME is empty, cannot pick a default unpack directory!

Nothing downloaded, packaged or imported.
```

Now using a similar recipe that passes in `destination_path` explicitly:
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>Description</key>
	<string>Recipe description</string>
	<key>Identifier</key>
	<string>local.TestExtractTwo</string>
	<key>Input</key>
	<dict/>
	<key>MiniumumVersion</key>
	<string>1.0</string>
	<key>Process</key>
	<array>
		<dict>
			<key>Arguments</key>
			<dict>
				<key>archive_path</key>
				<string>%PKG%</string>
				<key>destination_path</key>
				<string>%RECIPE_CACHE_DIR%/Test</string>
			</dict>
			<key>Processor</key>
			<string>Unarchiver</string>
		</dict>
	</array>
</dict>
</plist>
```

Running that recipe with a zip pkg:
```
$ ./autopkg run TestExtract2.extract -p ~/Downloads/test.zip
Processing TestExtract2.extract...
WARNING: TestExtract2.extract is missing trust info and FAIL_RECIPES_WITHOUT_TRUST_INFO is not set. Proceeding...

Nothing downloaded, packaged or imported.
$ ls ~/Library/AutoPkg/Cache/local.TestExtractTwo
Test     receipts
$ ls ~/Library/AutoPkg/Cache/local.TestExtractTwo/Test
TestZipContents.pdf
```

### CODE REVIEW

**Comment by gregneagle** [comment]
Nick: can you either fix the conflicts or close this PR?

**Comment by nmcspadden** [comment]
Retested, confirmed that it still works as described in the original test plan.

**Comment by nmcspadden** [comment]
Ping?

File: Code/autopkglib/Unarchiver.py [hunk 1]
```diff
@@ -85,10 +84,14 @@ def main(self):
...
-            os.path.join(self.env["RECIPE_CACHE_DIR"], self.env["NAME"]),
-        )
+        destination_path = self.env.get("destination_path")
+        if not destination_path and not self.env.get("NAME"):
+            raise ProcessorError(
+                "No destination path provided and NAME is empty, "
+                "cannot pick a default unpack directory!"
+            )
+        elif not destination_path:
+            os.path.join(self.env["RECIPE_CACHE_DIR"], self.env["NAME"])
```
Comment (gregneagle) [inline Code/autopkglib/Unarchiver.py]: Why not as a last resort, use `os.path.join(self.env["RECIPE_CACHE_DIR"], "unpack")` as the unpack directory?

File: Code/autopkglib/Unarchiver.py [hunk 2]
```diff
@@ -39,7 +38,7 @@ class Unarchiver(Processor):
     description = __doc__
     input_variables = {
         "archive_path": {
-            "required": False,
+            "required": True,
```
Comment (gregneagle) [inline Code/autopkglib/Unarchiver.py]: Why is this now required?

File: Code/autopkglib/Unarchiver.py [hunk 3]
```diff
@@ -81,15 +80,17 @@ def main(self):
         """Unarchive a file"""
         # handle some defaults for archive_path and destination_path
         archive_path = self.env.get("archive_path", self.env.get("pathname"))
```
Comment (gregneagle) [inline Code/autopkglib/Unarchiver.py]: Is archive_path required or not?

File: Code/autopkglib/Unarchiver.py [hunk 4]
```diff
@@ -81,15 +80,17 @@ def main(self):
...
-        if not archive_path:
-            raise ProcessorError(
-                "Expected an 'archive_path' input variable but none is set!"
```
Comment (gregneagle) [inline Code/autopkglib/Unarchiver.py]: Is archive_path required or not?

**Comment by nmcspadden** [comment]
Sorry, I confused myself. I forgot that it had a default value specified elsewhere when I set it to required of True.

**Pull request closed without merging**